Source/SwiftLintFramework/Helpers/BodyLineCounter.swift

```diff
@@ -0,0 +1,63 @@
+import SwiftSyntax
+
+/// Counts the number of lines in a scope's body.
+enum BodyLineCounter {
+    struct Result {
+        let exceeds: Bool
+        let lineCount: Int
+    }
+
+    /// This function determines if given a scope with a left/right brace, such as a function, closure, type, etc, if
+    /// the "body" of that scope meets or exceeds a specified line count limit when you ignore lines only containing
+    /// comments and/or whitespace.
+    ///
+    /// - parameter file:           The source file.
+    /// - parameter leftBraceLine:  The line where the scope's left opening brace is located.
+    /// - parameter rightBraceLine: The line where the scope's right closing brace is located.
+    /// - parameter limit:          The body line limit to stay under, ignoring lines only containing comments and/or
+    ///                             whitespace.
+    ///
+    /// - returns: A tuple containing whether the body `exceeds` the specific `limit` and `lineCount`, the number of
+    ///            effective lines of the body ignoring lines only containing comments and/or whitespace.
+    static func lineCountIgnoringCommentsAndWhitespace(
```

Comment:
Is there a convention on when to introduce a static function in a namespace and when to define it as a function in an extension (here an extension of `SwiftLintFile`)?

Reply:
No convention. I had this in an extension on `SwiftLintFile` in `SwiftLintFile+Regex.swift` at first but it wasn't related to regular expressions at all, so I wanted to move it elsewhere and came up with this namespace without thinking too deeply about it.

Now that you mention it, it's probably better as an extension on `SwiftLintFile` so we need fewer function parameters.

I'll move this to a `SwiftLintFile+BodyLineCount.swift` file.

Thanks for the idea!